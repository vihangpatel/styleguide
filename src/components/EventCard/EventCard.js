import React from 'react'
import PropTypes from 'prop-types'
import { createComponent } from 'react-fela'

import LazyImage from '../../common/ui/LazyImage'
import { noEventPreviewImageUrl, eventPreloaderUrl } from '../../common/config'
import { hrefHandler } from '../../utils/helpers'
import {
	__Card,
	__Image,
	__Price,
	__wrap,
	__blurimg,
	__imgWrap,
	__blurOverlay,
} from './__style'

// import all components if any
import DateBlock from './components/DateBlock'
import DetailBlock from './components/Detail'
import PlaceHolder from './components/PlaceHolder'

const WrapMain = createComponent(__Card)
const Image = createComponent(__Image)
const PriceInfo = createComponent(__Price)
const WrapDetail = createComponent(__wrap)
const BlurBanner = createComponent(__blurimg)
const ImageWrapper = createComponent(__imgWrap)
const BlurOverlay = createComponent(__blurOverlay)

const EventCard = props => {
	const { loading, lockWidth, height, ...otherProps } = props

	return (
		<div>
			<WrapMain lockWidth={lockWidth} height={height}>
				{loading ? (
					<DefaultCard {...otherProps} />
				) : (
					<DetailCard {...otherProps} />
				)}
			</WrapMain>
		</div>
	)
}

const DefaultCard = () => (
	<div>
		<PlaceHolder />
	</div>
)

const DetailCard = ({
	banner,
	bannerName,
	price,
	month,
	date,
	day,
	title,
	venue,
	type,
	href,
}) => {
	const isNonPriced = parseInt(price, 10) <= 0 || isNaN(price)

	return (
		<div
			onClick={hrefHandler(href)}
			onKeyDown={hrefHandler(href)}
			role="presentation"
		>
			<ImageWrapper>
				<BlurBanner>
					<LazyImage
						tinySrc={eventPreloaderUrl}
						src={banner}
						alt={bannerName || title}
						srcErr={noEventPreviewImageUrl}
					/>
					<BlurOverlay />
				</BlurBanner>
				<Image>
					<LazyImage
						tinySrc={eventPreloaderUrl}
						src={banner}
						alt={bannerName || title}
						srcErr={noEventPreviewImageUrl}
					/>
				</Image>
			</ImageWrapper>
			<WrapDetail isNonPriced={isNonPriced}>
				<DateBlock
					{...{
						month,
						date,
						day,
					}}
				/>
				<DetailBlock
					{...{
						title,
						venue,
						type,
					}}
				/>
			</WrapDetail>
			{!isNonPriced && <PriceInfo>&#8377; {price} Onwards</PriceInfo>}
		</div>
	)
}

EventCard.propTypes = {
	/** Whether to default the container width to 350px */
	lockWidth: PropTypes.bool,
	/** Height for the Eventcard */
	height: PropTypes.string,
	/** Gets called when the user clicks on the button */
	onClick: PropTypes.func,
	/** EventBanner Name for the event */
	bannerName: PropTypes.string,
	/** EventBanner Image for the event */
	banner: PropTypes.string,
	/** PriceInfo for the event */
	price: PropTypes.number,
	/** EventMonth for the event */
	month: PropTypes.string,
	/** EventDate for the event */
	date: PropTypes.string,
	/** EventDay for the event */
	day: PropTypes.string,
	/** EventTitle for the event */
	title: PropTypes.string,
	/** EventVenue for the event */
	venue: PropTypes.string,
	/** EventType for the event */
	type: PropTypes.string,
}

export default EventCard
