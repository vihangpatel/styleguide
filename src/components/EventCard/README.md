Basic Event Card:

```js
	<EventCard
			banner = '//in.bmscdn.com/webin/test-m6/event-card-test.jpg'
			price = {100}
			month = 'AUG'
			date = '05'
			day = 'Onwards'
			title = 'BookMyShow Festival'
			venue = 'Movie, Events, Play'
			type = 'Music, Electronic'
			lockWidth
	/>
```

Placeholder Event Card:

```js
	<EventCard loading lockWidth/>
```



