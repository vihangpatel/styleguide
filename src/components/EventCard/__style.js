import { margin, padding } from 'polished'

export const __Card = props => ({
	height: `${props.height ? props.height : '275'}`,
	overflow: 'hidden',
	border: '1px solid #cccccc',
	borderRadius: props.theme.borderRadius,
	background: 'white',
	...margin('0', 'auto', '10'),
	mobileXs: {
		width: `${props.lockWidth ? '290' : '100%'}`,
	},
	mobileX: {
		width: `${props.lockWidth ? '345' : '100%'}`,
	},
})

export const __Image = () => ({
	...margin('0', 'auto'),
	overflow: 'hidden',
	mobileXs: {
		height: '115',
	},
	mobileX: {
		height: '135',
	},
	'& img': {
		width: '100%',
		height: 'inherit',
	},
})

export const __Info = () => ({
	width: '100%',
	height: 'auto',
})

export const __Price = props => ({
	color: '#4A4A4A',
	fontSize: '12',
	textAlign: 'end',
	borderTop: '1px solid #F5F5FA',
	lineHeight: '3',
	...props.theme.mediumBaseFont,
	...margin('0', '15'),
})

export const __wrap = props => ({
	display: 'flex',
	alignItems: 'flex-start',
	...padding('15', '0', '0'),
	mobileXs: {
		height: '120',
	},
	mobileX: {
		height: '100',
	},
	extend: {
		condition: props.isNonPriced === true,
		style: {
			paddingTop: 25,
		},
	},
})

export const __DateBlock = theme => ({
	textAlign: 'center',
	width: '60',
	color: '#333333',
	verticalAlign: 'top',
	'& div': {
		...theme.mediumBaseFont,
		fontSize: '12',
	},
	'& ._month': {
		color: '#4CD9AA',
		...theme.boldBaseFont,
		textTransform: 'uppercase',
	},
	'& ._date': {
		fontSize: '18',
		lineHeight: '27px',
		...theme.boldBaseFont,
	},
})

export const __DetailBlock = theme => ({
	width: '75%',
	...padding(0, 10),
	'& div': {
		fontSize: '12',
	},
	'& ._title': {
		fontSize: '14',
		color: theme.color.charcoal,
		marginBottom: '10',
		overflow: 'hidden',
		...theme.mediumBaseFont,
		display: '-webkit-box',
		'-webkit-line-clamp': 2,
		textOverflow: 'ellipsis',
		'-webkit-box-orient': 'vertical',
		'&': {
			display: '-moz-box',
		},
		mobileXs: {
			maxHeight: 34,
		},
		mobileX: {
			maxHeight: 40,
		},
	},
	'& ._venue': {
		color: theme.colorGrayBorder,
		marginBottom: 5,
		maxHeight: 15,
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		display: '-webkit-box',
		'-webkit-line-clamp': 1,
		'-webkit-box-orient': 'vertical',
		...theme.regularBaseFont,
	},
	'& ._type': {
		color: theme.colorGrayBorder,
		maxHeight: 15,
		overflow: 'hidden',
		marginBottom: 5,
		display: '-webkit-box',
		'-webkit-line-clamp': 1,
		textOverflow: 'ellipsis',
		'-webkit-box-orient': 'vertical',
		...theme.regularBaseFont,
	},
})

export const __imgWrap = () => ({
	position: 'relative',
	overflow: 'hidden',
})

export const __blurimg = () => ({
	position: 'absolute',
	left: '0',
	right: '0',
	top: '0',
	bottom: '0',
	filter: 'blur(2vw)',
	transform: 'scale(1.05)',
	backgroundSize: '100%',
	'& img': {
		width: '100%',
		height: '100%',
	},
})

export const __blurOverlay = () => ({
	backgroundImage:
		'linear-gradient(to bottom,rgba(0,0,0,.08) 46%,rgba(0,0,0,.08) 59%,rgba(0,0,0,.9) 100%)',
	backgroundSize: '100%',
	position: 'absolute',
	left: '0',
	right: '0',
	top: '0',
	bottom: '0',
})
