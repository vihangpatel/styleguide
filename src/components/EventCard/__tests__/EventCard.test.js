import React from 'react'
import EventCard from 'components/EventCard/EventCard'
import StyleProvider from 'utils/Provider'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'

describe('EventCard Testing', () => {
	it('renders correctly', () => {
		const eventCard = renderer
			.create(
				<StyleProvider>
					<EventCard
						banner="//in.bmscdn.com/webin/test-m6/event-card-test.jpg"
						price="`100 onwards"
						month="AUG"
						date="05"
						day="Onwards"
						title="BookMyShow Festival"
						venue="Movie, Events, Play"
						type="Music, Electronic"
						lockWidth
					/>
				</StyleProvider>
			)
			.toJSON()

		expect(eventCard).toMatchSnapshot()
	})

	it('renders correctly with enzyme', () => {
		const renderedComponent = shallow(
			<StyleProvider>
				<EventCard
					banner="//in.bmscdn.com/webin/test-m6/event-card-test.jpg"
					price="`100 onwards"
					month="AUG"
					date="05"
					day="Onwards"
					title="BookMyShow Festival"
					venue="Movie, Events, Play"
					type="Music, Electronic"
					lockWidth
				/>
			</StyleProvider>
		)
		// TODO: Needs research
		expect(true).toBe(true)
	})

	it('SHould be clickable', () => {
		const clickSpy = jest.fn()
		const renderedComponent = mount(
			<StyleProvider>
				<EventCard
					banner="//in.bmscdn.com/webin/test-m6/event-card-test.jpg"
					price="`100 onwards"
					month="AUG"
					date="05"
					day="Onwards"
					title="BookMyShow Festival"
					venue="Movie, Events, Play"
					type="Music, Electronic"
					lockWidth
					onClick={clickSpy}
				/>
			</StyleProvider>
		)
		renderedComponent.simulate('click')
		expect(clickSpy).toHaveBeenCalled()
	})
})
