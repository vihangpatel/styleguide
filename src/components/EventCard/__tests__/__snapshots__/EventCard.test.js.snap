// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Testing renders correctly 1`] = `
<div>
  <div
    className="a b c d e f g h i j k"
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      <div
        className="l b"
      >
        <div
          className="m n o p q r s t u v"
        >
          <div>
            <img
              alt="BookMyShow Festival"
              className="w cy y z ab n p cz da db dc dd"
              onClick={undefined}
              src="//in.bmscdn.com/webin/test-m6/event-card-test.jpg"
            />
          </div>
          <div
            className="ae t m n o p q"
          />
        </div>
        <div
          className="f g af i b u ag ah ai"
        >
          <div>
            <img
              alt="BookMyShow Festival"
              className="w cy y z ab n p cz da db dc dd"
              onClick={undefined}
              src="//in.bmscdn.com/webin/test-m6/event-card-test.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="aj ak al am an ao ap aq ar"
      >
        <div
          className="as at au av aw ax ay az ba bb bc bd be bf"
        >
          <div
            className="_month"
          >
            AUG
          </div>
          <div
            className="_date"
          >
            05
          </div>
          <div
            className="_day"
          >
            Onwards
          </div>
        </div>
        <div
          className="bg bh am an ao ay bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by bz ca cb cc cd ce cf cg ch ci cj ck cl cm cn co"
        >
          <div
            className="_title"
          >
            BookMyShow Festival
          </div>
          <div
            className="_venue"
          >
            Movie, Events, Play
          </div>
          <div
            className="_type"
          >
            Music, Electronic
          </div>
        </div>
      </div>
      <div
        className="cp cq cr cs ct cu cv f cw af cx"
      >
        â‚¹ 
        \`100 onwards
         Onwards.
      </div>
    </div>
  </div>
</div>
`;
